src/gtfs/read.rs

```diff
@@ -576,6 +576,17 @@ where
 {
     let filename = "agency.txt";
     let gtfs_agencies = read_objects::<_, Agency>(file_handler, filename)?;
+
+    for (current_agency, next_agency) in gtfs_agencies.iter().zip(gtfs_agencies.iter().skip(1)) {
+        if current_agency.timezone != next_agency.timezone {
+            warn!(
+                "different agency timezone: {} - {}",
+                current_agency.timezone, next_agency.timezone
+            );
+            break;
+        }
+    }
```

Comment:
How about something like this?

```suggestion
    let timezones: HashSet<_> = gtfs_agencies.iter().map(|agency| agency.timezone).collect();
    if timezones.len() > 1 {
        if current_agency.timezone != next_agency.timezone {
            warn!(
                "different agency timezone: {} - {}",
                current_agency.timezone, next_agency.timezone
            );
            break;
        }
    }
```